src/quo2/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,295 @@
+(ns quo2.components.wallet.transaction-progress.view
+  (:require [quo2.components.wallet.transaction-progress.style :as style]
+            [quo2.core :as quo2]
+            [quo2.components.selectors.selectors.view :as selectors]
+            [quo2.components.buttons.button.view :as button]
+            [quo2.components.wallet.progress-bar.view :as progress-box]
+            [quo2.components.markdown.text :as text]
+            [quo2.components.tags.status-tags :as status-tag]
+            [quo2.foundations.colors :as colors]
+            [status-im2.common.resources :as resources]
+            [quo2.theme :as theme]
+            [reagent.core :as reagent]
+            [react-native.core :as rn]))
+
+(defn get-colors
+  [color]
+  (let [current-theme (theme/get-theme)]
+    (cond
+      (= current-theme :dark)
+      (cond
+        (= color "neutral-5")  colors/neutral-70
+        (= color "neutral-10") colors/neutral-80
+        (= color "neutral-40") colors/neutral-50
+        (= color "neutral-50") colors/neutral-60
+        (= color "danger-50")  colors/danger-60
+        (= color "success-50") colors/success-60)
+      :else
+      (cond
+        (= color "neutral-5")  colors/neutral-5
+        (= color "neutral-10") colors/neutral-10
+        (= color "neutral-40") colors/neutral-40
+        (= color "neutral-50") colors/neutral-50
+        (= color "danger-50")  colors/danger-50
+        (= color "success-50") colors/success-50))))
+
+(defn load-icon
+  [icon color]
+  [rn/view {:style style/icon}
+   [quo2/icon icon
+    {:color color}]])
+
+(def total-box 85)
+
+(def app-state (reagent/atom {:counter 0}))
+(def interval-id (reagent/atom nil)) ; To store the interval ID
+
+(defn stop-interval
+  []
+  (when @interval-id
+    (js/clearInterval @interval-id)
+    (reset! interval-id nil))) ; Clear the interval ID
+
+(defn clear-counter
+  []
+  (swap! app-state assoc :counter 0))
+
+(defn update-counter
+  [networkState]
+  (let [new-counter-value (-> @app-state :counter inc)]
+    (if (or (and (= networkState :pending) (> new-counter-value 1))
+            (and (= networkState :sending) (> new-counter-value 2))
+            (and (= networkState :confirmed) (> new-counter-value 4))
+            (and (= networkState :finalising) (> new-counter-value 18))
+            (and (= networkState :finalized) (> new-counter-value total-box))
+            (and (= networkState :error) (> new-counter-value 2)))
+      (stop-interval)
+      (swap! app-state assoc :counter new-counter-value))))
+
+(defn start-interval
+  [networkState]
+  (reset! interval-id
+    (js/setInterval
+     (fn []
+       (update-counter networkState))
+     50))) ; Interval of 50ms
```

Comment:
imo, better to put the interval in a named const outside this function and remove this comment as the variable will describe it's use.

Reply:
got it